### ISSUE TITLE
Restart nexus433 when Home Assistant is restarted

### ISSUE THREAD

**Issue by ktuulos** [aquaticus/nexus433#32]
Nexus433 announces sensors to Home Assistant only, when sensor is found. This means, that if Home Assistant is restarted, it does not know anything about sensors. I have had to manually restart nexus433 after restarting Home Assistant in order to get sensors announced again.

For automating this, I have created following:

1) Line in crontab:
`@reboot ~/work/nexus-restart-subscribe.sh`

2) Contents of the first script:
```
pi@nexus433:~ $ cat work/nexus-restart-subscribe.sh
#!/bin/bash
sleep 60
mosquitto_sub -t "homeassistant/status" -h <MQTT_BROKER_IP> | xargs -n 1 ~/work/nexus-restart.sh
```

3) Contents of the second script:
```
pi@nexus433:~ $ cat work/nexus-restart.sh
#!/bin/bash
if [ "$1" == "online" ]
then
  sudo systemctl restart nexus433.service
fi

```

This is a quick and dirty trick to enable nexus433 automatically restart, when Home Assistant instance is restarted.

Could this behavior be integrated into nexus433 itself?

**Comment by aquaticus** [comment]
Hi,
thanks for your comment.

What do you mean by "it does not know anything about sensors"? Are you talking about HA automatic sensor discovery? I was sure HA keeps info about discovered sensors between restarts (but can be wrong I tested this with really outdated version).

**Comment by ktuulos** [comment]
Yes, the HA automatic MQTT sensor discovery. I'm using HA under supervised installation, the current version is 2021.9.7.

**Comment by aquaticus** [comment]
thanks for clarification.

This seems like a serious bug. The entire discovery functionality is usless. I try to fix that.

**Comment by ktuulos** [comment]
When you are fixing that, could you also add unique IDs to reported sensors?

I have added those to my own installation, to MQTTClient::SensorDiscover -function, by adding unique_id reporting. The unique_id was just the name added with type.

The lack of unique IDs makes it impossible to rename sensors in HA.

**Comment by aquaticus** [comment]
yeah, sure good point.

**Comment by ktuulos** [comment]
Here is reference implementation for this later change. Note that I also changed the device class for "quality" from None to "signal_strength".

```
void MQTTClient::SensorDiscover(uint16_t id)
{
    Substitute(id);

    const char* topic_fmt[4] =
    {
            "%s/sensor/" NEXUS433 "_%04x_temperature/config",
            "%s/sensor/" NEXUS433 "_%04x_humidity/config",
            "%s/sensor/" NEXUS433 "_%04x_battery/config",
            "%s/sensor/" NEXUS433 "_%04x_quality/config"
    };

    const char* units[4] = { "°C", "%", "%", "%" };
    const char* value[4] = { "temperature", "humidity", "battery", "quality" };
    const char* device_class[4] = { "temperature", "humidity", "battery", "signal_strength" };

    std::string name;
    char topic[256];
    char payload[512];
    char state_topic[256];

    for (int i = 0; i < 4; i++)
    {
        snprintf(topic, sizeof(topic), topic_fmt[i], Config::transmitter::discovery_prefix.c_str(), id);
        name = GetName((TYPES)i, id);
        snprintf(state_topic, sizeof(state_topic), m_StateTopic, id);

        snprintf(payload, sizeof(payload),
                "{"
                "\"name\": \"%s\", "
                "\"device_class\": \"%s\", "
                "\"state_topic\": \"%s\", "
                "\"availability_topic\": \"" NEXUS433 "/connection\", "
                "\"unit_of_measurement\": \"%s\", "
                "\"value_template\": \"{{ value_json.%s }}\", "
                "\"expire_after\": %d, "
                "\"unique_id\": \"nexus433_%04x_%s\""
                "}",
                name.c_str(),
                device_class[i],
                state_topic,
                units[i],
                value[i],
                PacketStorage::GetSilentTimeoutSec(),
                id,
                value[i]);

        // publish
        publish(NULL, topic, strlen(payload), payload);
    }
}

```

**Comment by aquaticus** [comment]
Fixed in 07b4182
